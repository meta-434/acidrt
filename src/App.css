.app {
  text-align: center;
  background-color: #a3ad71;
  display: flex;
  flex-direction: column;
}

main {
  min-height: calc(100vh - 350px);
}
