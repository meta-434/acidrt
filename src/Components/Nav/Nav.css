nav {
    display: flex;
    flex-direction: column;
    text-align: center;
    background-color: #ACC8E5;
    padding: 10px;
    max-height: 250px;
    min-height: 250px;
}

h1 {
    font-size: xx-large;
}

button {
    background-color: #ACC8E5;
    border-radius: 5px;
    border: 2px solid #112A46;
    margin: 0 2px 0 2px;
}

a {
    text-decoration: none;
    color: #112A46;
}

.title-header {
    color: #112A46;

}

@media only screen and (min-width: 600px) {
    nav {
        flex-direction: row;
        text-align: left;
    }

    nav > * {
        margin: auto;
    }
}