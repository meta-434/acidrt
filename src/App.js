import React, { Component } from "react";
import "./App.css";
import Nav from "./Components/Nav/Nav.js";
import Login from "./Components/Login/Login.js";
import Main from "./Components/Main/Main.js";
import Form from "./Components/Form/Form.js";
import Dashboard from "./Components/Dashboard/Dashboard";
import ReportList from "./Components/ReportList/ReportList.js";
import ReportDisplay from "./Components/ReportDisplay/ReportDisplay.js";
import AcidrtContext from "./AcidrtContext";
import { Route } from "react-router-dom";
import Footer from "./Components/Footer/Footer";

class App extends Component {
  state = {
    reports: [],
    authToken: undefined,
    error: undefined,
    username: undefined,
    password: undefined,
  };

  componentDidMount() {
    this.handleGetErrors();
  }

  handleGetErrors() {
    this.setState({ error: this.context.error });
  }

  clearError = () => {
    this.setState({ error: undefined });
  };

  handleLogOut = () => {
    this.setState({ authToken: undefined });
    sessionStorage.clear();
  };

  handlePostAuthenticate = ({ username, password }) => {
    return fetch(process.env.REACT_APP_SERVER_URL + `/authenticate`, {
      method: "post",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        username: username,
        password: password,
      }),
    })
      .then((result) => {
        return result.json();
      })
      .then((resJson) => {
        this.setState({ error: resJson });
        if (!!resJson.token) {
          this.setState({ authToken: resJson.token, username });
          sessionStorage.setItem("access-token", resJson.token);
          sessionStorage.setItem("username", username);
          this.handleGetReports();
        } else {
          throw new Error(
            " error in authenticating. check username and password. "
          );
        }
      })
      .catch((error) => {
        console.error(error);
      });
  };

  handlePostReport = (newReport) => {
    return fetch(process.env.REACT_APP_SERVER_URL + `/submit/`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(newReport),
    })
      .then((res) => res.json())
      .catch((error) => {
        console.error("error: ", error);
      });
  };

  handleGetUniqueReport = (id) => {
    return fetch(process.env.REACT_APP_SERVER_URL + `/api/reports/${id}`, {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        "access-token": `${
          sessionStorage[`access-token`] || this.state.authToken
        }`,
      },
    })
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.success && responseJson.success === false) {
          throw new Error(`error in getting report ${id}`);
        } else {
          return responseJson;
        }
      })
      .catch((error) => {
        console.error(error);
        this.setState({ error });
      });
  };

  handleGetReports = async () => {
    return fetch(process.env.REACT_APP_SERVER_URL + `/api/reports`, {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        "access-token": `${
          sessionStorage[`access-token`] || this.state.authToken
        }`,
      },
    })
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.success && responseJson.success === false) {
          throw new Error("error in getting reports");
        } else {
          this.setState({
            reports: responseJson,
          });
        }
      })
      .catch((error) => {
        console.error(error);
        this.setState({ error });
      });
  };

  handleDeleteReport = (reportId) => {
    return fetch(
      process.env.REACT_APP_SERVER_URL + `/api/reports/${reportId}`,
      {
        method: "delete",
        headers: {
          "Content-Type": "application/json",
          "access-token": `${
            this.state.authToken || sessionStorage[`access-token`]
          }`,
        },
      }
    )
      .then((res) => this.handleGetReports())
      .catch((error) => {
        console.error(error);
        this.setState({ error });
      });
  };

  handleEditReport = (reportId, updatedReport) => {
    return fetch(
      process.env.REACT_APP_SERVER_URL + `/api/reports/${reportId}`,
      {
        method: "put",
        headers: {
          "Content-Type": "application/json",
          "access-token": `${
            this.state.authToken || sessionStorage[`access-token`]
          }`,
        },
        body: JSON.stringify(updatedReport),
      }
    )
      .then((res) => {
        if (res && res.ok) {
          return res;
        } else {
          throw new Error("Error in updating report");
        }
      })
      .then((res) => res.json())
      .then((resJson) => {
        this.handleGetReports();
        return resJson;
      })
      .catch((error) => console.error("error", error));
  };

  render() {
    const { reports, authToken, username, error } = this.state;
    const context = {
      reports,
      username,
      authToken,
      error,
      handlePostAuthenticate: this.handlePostAuthenticate,
      handlePostReport: this.handlePostReport,
      handleGetReports: this.handleGetReports,
      handleGetUniqueReport: this.handleGetUniqueReport,
      handleDeleteReport: this.handleDeleteReport,
      handleEditReport: this.handleEditReport,
      handleLogOut: this.handleLogOut,
      handleGetErrors: this.handleGetErrors,
      handleLatLng: this.handleLatLng,
      clearError: this.clearError,
    };
    return (
      <AcidrtContext.Provider value={context}>
        <div className="app">
          <Nav />
          <main className="app-content">
            <Route exact path="/" render={(props) => <Main {...props} />} />
            <Route path="/login" render={(props) => <Login {...props} />} />
            <Route path="/form" render={(props) => <Form {...props} />} />
            <Route
              exact
              path="/all-reports"
              render={(props) => <ReportList {...props} />}
            />
            <Route
              path="/dashboard"
              render={(props) => <Dashboard {...props} />}
            />
            <Route
              path="/all-reports/:reportId"
              render={(props) => <ReportDisplay {...props} />}
            />
          </main>
          <Footer />
        </div>
      </AcidrtContext.Provider>
    );
  }
}

export default App;
