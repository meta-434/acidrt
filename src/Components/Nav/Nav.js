import React, { Component } from 'react';
import AcidrtContext from "../../AcidrtContext";
import { Link } from "react-router-dom";
import './Nav.css';

class Nav extends Component {
    static contextType = AcidrtContext;

    logOut = () => {
        this.context.handleLogOut();
        // this.props.history.push('/');
        window.location.href = '/'
    }

    render() {
        return(
            <nav>
                <section className={'title-header'}>
                    <header>
                        <h1 className="nav-h1"><Link to='/'>ACIDRT</Link></h1>
                        <p><b>A</b>lbemarle <b>C</b>ounty <b>I</b>llicit <b>D</b>ischarge <b>R</b>eporting <b>T</b>ool</p>
                    </header>
                </section>
                <section className={'nav-buttons'}>
                    {(!!this.context.authToken || sessionStorage[`access-token`])
                        ? <>
                            <p>Welcome, {sessionStorage[`username`] || this.context.username}</p>
                            <button><Link to='/form'>Form</Link></button>
                            <button><Link to='/all-reports/'>All Reports</Link></button>
                            <button><Link to='/dashboard'>Dashboard</Link></button>
                            <button onClick={this.logOut}>Log Out</button>
                        </>
                        : <>
                            <button><Link to='/form'>Form</Link></button>
                            <button><Link to='/login'>Log In</Link></button>
                        </>
                    }
                </section>
            </nav>
        );
    }
}

export default Nav;