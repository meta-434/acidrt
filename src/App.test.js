import React from "react";
import App from "./App";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Nav from "./Components/Nav/Nav";

test("renders the app landing page", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
