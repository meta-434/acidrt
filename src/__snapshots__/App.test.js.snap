// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app landing page 1`] = `
<div
  className="app"
>
  <nav>
    <section
      className="title-header"
    >
      <header>
        <h1
          className="nav-h1"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            ACIDRT
          </a>
        </h1>
        <p>
          <b>
            A
          </b>
          lbemarle 
          <b>
            C
          </b>
          ounty 
          <b>
            I
          </b>
          llicit 
          <b>
            D
          </b>
          ischarge
           
          <b>
            R
          </b>
          eporting 
          <b>
            T
          </b>
          ool
        </p>
      </header>
    </section>
    <section
      className="nav-buttons"
    >
      <button>
        <a
          href="/form"
          onClick={[Function]}
        >
          Form
        </a>
      </button>
      <button>
        <a
          href="/login"
          onClick={[Function]}
        >
          Log In
        </a>
      </button>
    </section>
  </nav>
  <main
    className="app-content"
  >
    <main>
      <section>
        <header
          role="banner"
        >
          <p
            className="login-note"
          >
            <em>
              For testing authorized user functionality, use:
              <br />
              username: acps-employee 
              <br />
              password: ilovemywatershed
            </em>
          </p>
          <h1>
            Albemarle County / Rivanna Stormwater
          </h1>
          <h2>
            Illicit Discharge Reporting
          </h2>
        </header>
      </section>
      <section>
        <header>
          <h3>
            Stormdrains are designed to handle stormwater runoff
             
            <em>
              only!
            </em>
          </h3>
        </header>
        <p>
          They do not provide treatment to clean and purify the water or anything else that enters storm sewer systems. The City of Charlottesville, the County of Albemarle, and the University of Virginia manage the storm drain systems throughout our community.
        </p>
        <br />
        <p>
          As part of their Illicit Discharge Detection and Elimination (IDDE) programs, the members of RSEP have set up a hot line number for citizens to call (434-202-4179) if they suspect they have witnessed illicit dumping into or near a storm drain, or other potential water pollution incidents. If you are not sure whether an illicit discharge has occurred, we suggest you report it anyway, and we can determine whether some action is required to protect the water supply and/or the environment.
        </p>
      </section>
      <section>
        <header>
          <h3>
            How to report a pollution problem?
          </h3>
        </header>
        <p>
          To report a suspected illicit discharge, complete the online reporting form or call 434-202-4179. This is not an emergency response number. To report an emergency, please call 911.
        </p>
        <p>
          <em>
            <b>
              REMEMBER: You don't need an account to submit a report!
            </b>
          </em>
        </p>
      </section>
      <section>
        <header>
          <h3>
            What are Some Examples of Illicit Discharges?
          </h3>
        </header>
        <p>
          Proper disposal of common household chemicals and products is important to prevent pollution of our local waterways. Under no circumstances should anything be disposed of down a storm drain or directly into a local drainageway. The following is a list of common sources of illicit discharges:
        </p>
        <ul>
          <li>
            “Gray Water” such as laundry, shower, or kitchen waste water.
          </li>
          <li>
            Paints, solvents, cleaning products and other toxic household substances
          </li>
          <li>
            Gasoline, motor oil, brake fluid, engine coolant (antifreeze) and other toxic automotive products.
          </li>
          <li>
            Septic tank pumpout and other types of human or animal waste.
          </li>
          <li>
            Fertilizers, pesticides, herbicides and other agricultural products.
          </li>
          <li>
            Cooking oil or grease.
          </li>
        </ul>
      </section>
      <section>
        <header>
          <h3>
             
            What are Some Indications that an Illicit Discharge Has Occurred?
          </h3>
        </header>
        <ul>
          <li>
            An unnatural color in the water.
          </li>
          <li>
            An oil sheen floating on the water.
          </li>
          <li>
            Floatables such as toilet paper or suds.
          </li>
          <li>
            A foul or unnatural odor.
          </li>
          <li>
            Dead fish or other aquatic creatures.
          </li>
        </ul>
      </section>
    </main>
  </main>
  <div
    className="footer-div"
  >
    <button>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </button>
    <p>
      made by
       
      <a
        href="http://alex-hapgood.info"
        rel="noopener noreferrer"
        target="_blank"
      >
        Alex Hapgood
      </a>
       
      for Albemarle County Facilities and Environmental Stewardship Department
    </p>
    <p>
      updated 8/2020
    </p>
  </div>
</div>
`;
